src/pages/storage/forms/StoragePoolClusteredSourceSelector.tsx

```diff
@@ -0,0 +1,41 @@
+import { Label } from "@canonical/react-components";
+import ClusterSpecificInput from "components/forms/ClusterSpecificInput";
+import { FormikProps } from "formik";
+import { FC } from "react";
+import { ensureEditMode } from "util/instanceEdit";
+import { StoragePoolFormValues } from "./StoragePoolForm";
+import { useClusterMembers } from "context/useClusterMembers";
+import { focusField } from "util/formFields";
+
+interface Props {
+  formik: FormikProps<StoragePoolFormValues>;
+}
+
+const StoragePoolClusteredSourceSelector: FC<Props> = ({ formik }) => {
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const memberNames: string[] = [];
+  clusterMembers.forEach((member) => memberNames.push(member.server_name));
+
+  return (
+    <>
+      <Label>Source</Label>
+      <ClusterSpecificInput
```

Comment:
We should also disable the `Same for all cluster members` checkbox in case of the edit form.

And I think we should preserve the help text `Source can't be changed` below the last input in the edit case.

Reply:
Or maybe better, set `canToggleSpecific` to false in case of editing, so the checkbox is not even visible.